composer.json

```diff
@@ -17,7 +17,7 @@
 		"ext-json": "*"
 	},
 	"require-dev": {
-		"phpunit/phpunit": "^7.5 || ^8.0",
+		"phpunit/phpunit": "^9.0",

```

Comment:
just for sanity, does this upgrade work for both php versions (i.e. 8.0 and 7.1)?

Reply:
Thanks for calling this out.  PHPUnit needs a minimum of `7.3`. Also, `php 7.1` is no longer supported. Thinking we should consider upgrading to a minimum of `php 7.4`.

Reply:
Added a commit to bump this up.

Reply:
Unfortunately 7.3 to 7.4 is not a backwards compatible upgrade: https://www.php.net/manual/en/migration74.incompatible.php

I am assuming 7.1 to 7.4 will have more incompatibilities. @MIchaelMainer, do we have any community feedback regarding which version we should be on?

@Ndiritu alternatively, could we do something like: `^8.0 || ^9.0` for phpunit?

Reply:
@zengin , makes sense. Added the option for phpunit ^8.0. However this needs a minimum of php 7.2. I've updated the minimum php version for the SDK to php 7.2 for now. Confirmed that unit tests pass.

Reply:
@Ndiritu thanks for making the changes. There are still incompatibilities between 7.1 and 7.2: https://www.php.net/manual/en/migration72.incompatible.php

Given that it is a relatively older minor version we are upgrading to, I am assuming that most customers will already be there. We need at least a minor version bump for this.

To me, any breaking change needs to go out with a major bump according to [SemVer](https://semver.org/). Here is a nice bit on that topic: https://semver.org/#if-even-the-tiniest-backwards-incompatible-changes-to-the-public-api-require-a-major-version-bump-wont-i-end-up-at-version-4200-very-rapidly

That said, I think we can align with PHP's philosophy here which allows minor breaking changes. @MIchaelMainer, what do you think?

Reply:
@zengin Thanks for highlighting incompatibilities between PHP 7.1 and 7.2

More thoughts... @MIchaelMainer @zengin @SilasKenneth @roinochieng 
Was looking at the most recent PHP version usage metrics on Packagist [here](https://blog.packagist.com/php-versions-stats-2020-1-edition/). Given that most packages still cover php 7.1 and majority usage is only at php 7.3. I'm thinking the impact of this change to 7.2 should warrant a major version upgrade.

Reply:
We should move forward with the LTS versions of PHP and target minimum at 7.3, with plans for targeting 7.4 when 7.3 is no longer in LTS.

If a customer wants to stay with 7.1, which is out of LTS, here are the options they have:

1. Upgrade to 7.3 and do the work to support it in the case there are incompatibilities. 
2. In general, the most likely APIs to be adopted are the ones that have been implemented earlier. So, our older SDK versions will likely cover most integration scenarios encountered by our customers. We should keep our eyes out for scenarios where customers are stuck in an older PHP version but need new versions of generated models - if we encounter  this, then we can reevaluate.
3. In the case that a customer is pegged to 7.1, and needs to use a new model not generated for a 7.1 version of the client library, we have a mechanism for them to use the raw request and response. This is a supported scenario and was implemented for this case. They could implement their own model as well.

Reply:
@zengin The only community feedback I recall is that we should target the latest supported versions of PHP.

Reply:
@zengin Upgraded minimum php version to 7.3 and bumped up the SDK minimum version